src/V3Randomize.cpp

```diff
@@ -933,6 +1202,58 @@ class RandomizeVisitor final : public VNVisitor {
+            UASSERT_OBJ(randModeTarget.classp, nodep,
+                        "Should have checked in RandomizeMarkVisitor");
+            AstVar* const randModeVarp = getRandModeVar(randModeTarget.classp);
+            AstNodeExpr* lhsp = nullptr;
+            if (randModeTarget.classp == m_modp) {
+                // Called on 'this' or a member of 'this'
+                lhsp = new AstVarRef{nodep->fileline(), VN_AS(randModeVarp->user2p(), NodeModule),
+                                     randModeVarp, VAccess::WRITE};
+            } else {
+                AstMemberSel* const memberselp = new AstMemberSel{
+                    nodep->fileline(), randModeTarget.fromp->unlinkFrBack(), randModeVarp};
+                memberselp->foreach([](AstVarRef* varrefp) { varrefp->access(VAccess::WRITE); });
+                lhsp = memberselp;
+            }
+            if (nodep->pinsp()) {  // Set rand mode
+                UASSERT_OBJ(VN_IS(nodep->backp(), StmtExpr), nodep, "Should be a statement");
+                AstNodeExpr* const rhsp = VN_AS(nodep->pinsp(), Arg)->exprp()->unlinkFrBack();
+                if (randModeTarget.receiverp && randModeTarget.receiverp->isRand()) {
+                    // Called on a rand member variable. Set the variable's rand mode
+                    const VarRandMode randMode = {.asInt = randModeTarget.receiverp->user1()};
+                    UASSERT_OBJ(randMode.usesRandMode, nodep, "Failed to set usesRandMode");
+                    AstCMethodHard* const atp
+                        = new AstCMethodHard{nodep->fileline(), lhsp, "at",
+                                             new AstConst{nodep->fileline(), randMode.index}};
+                    atp->dtypeSetUInt32();
+                    m_stmtp->replaceWith(new AstAssign{nodep->fileline(), atp, rhsp});
+                } else {
+                    // Called on 'this' or a non-rand class instance. Set the rand mode of all
+                    // members
+                    m_stmtp->replaceWith(
+                        makeRandModeInitLoop(nodep->fileline(), lhsp, rhsp, m_ftaskp));
+                }
+                pushDeletep(m_stmtp);
+            } else {  // Retrieve rand mode
+                UASSERT_OBJ(randModeTarget.receiverp, nodep, "Should have receiver");
+                UASSERT_OBJ(randModeTarget.receiverp->isRand(), nodep, "Should be rand");
+                const VarRandMode randMode = {.asInt = randModeTarget.receiverp->user1()};
+                UASSERT_OBJ(randMode.usesRandMode, nodep, "Failed to set usesRandMode");
+                AstCMethodHard* const atp
+                    = new AstCMethodHard{nodep->fileline(), lhsp, "at",
+                                         new AstConst{nodep->fileline(), randMode.index}};
+                atp->dtypeSetUInt32();
+                nodep->replaceWith(atp);
+                VL_DO_DANGLING(pushDeletep(nodep), nodep);
+            }
+            return;
```

Comment:
```suggestion
```

Reply:
This is actually needed, it's an early return from this function.